src/main/java/org/quiltmc/asmr/processor/AsmrProcessorRunner.java

```diff
@@ -462,36 +435,27 @@ private void shiftRefCapturesFrom(List<AsmrReferenceCapture> sortedRefCaptures,
                 int[] pathPrefixB = refB.capture.pathPrefix();
 
                 // check that the b prefix starts with the a prefix
-                if (pathPrefixB.length < pathPrefixA.length) {
+                int bLength = pathPrefixB.length;
+                int aLength = pathPrefixA.length;
+
+                if (bLength < aLength) {
                     break bLoop;
                 }
-                for (int i = pathPrefixA.length - 1; i >= 0; i--) {
+
+                for (int i = 0; i < aLength; i++) {
                     if (pathPrefixB[i] != pathPrefixA[i]) {
                         break bLoop;
                     }
                 }
 
-                // check b isn't past the end of a if a is a capture
-                if (refA.capture instanceof AsmrReferenceSliceCapture) {
-                    AsmrReferenceSliceCapture<?, ?> sliceA = (AsmrReferenceSliceCapture<?, ?>) refA.capture;
-                    if (pathPrefixB.length > pathPrefixA.length) {
-                        if (pathPrefixB[pathPrefixA.length] >= sliceA.endIndexExclusive()) {
-                            break bLoop;
-                        }
-                    } else { // if (pathPrefixA.length == pathPrefixB.length)
-                        if (refB.capture instanceof AsmrReferenceSliceCapture) {
-                            AsmrReferenceSliceCapture<?, ?> sliceB = (AsmrReferenceSliceCapture<?, ?>) refB.capture;
-                            if (sliceB.startVirtualIndex() >= sliceA.endVirtualIndex()) {
-                                break bLoop;
-                            }
-
-                            // special case: check if b is an empty slice on the edge of a, then it's not colliding
-                            if (sliceB.startVirtualIndex() == sliceB.endVirtualIndex()
-                                    && (sliceA.startVirtualIndex() == sliceB.startVirtualIndex() || sliceA.endVirtualIndex() == sliceB.startVirtualIndex())) {
-                                continue bLoop;
-                            }
-                        }
-                    }
+                // check b isn't past the end of a
+                AsmrReferenceCapture aCapture = refA.capture;
+                AsmrReferenceCapture bCapture = refB.capture;
+
+                int bStartPathAtEndOfA = (bLength > aLength) ? pathPrefixB[aLength] : bCapture.startVirtualIndex();
+
+                if (bStartPathAtEndOfA >= aCapture.endIndexExclusive()) {
```

Comment:
I think `bCapture.startIndexInclusive()` instead should be valid here.

Reply:
I think the correct logic is to compare indexes if `bLength > aLength`, and compare virtual indexes otherwise. The updated code will incorrectly determine two slices as non-overlapping if `a.endVirtualIndex == 3` and `b.startVirtualIndex == 2`, for example.